splice_machine/src/test/java/com/splicemachine/benchmark/FeatureStoreBenchmark.java

```diff
@@ -0,0 +1,477 @@
+        for (int i = 0; i < PK.length - 1; ++i) {
+            PK[i] = (int)(id % 64);
+            id /= 64;
+        }
+        PK[PK.length - 1] = (int)id;
+    }
+
+    private static void populateTable(TableConfig config) {
+        try (Connection conn = makeConnection()) {
+            StringBuilder sb = new StringBuilder();
+            sb.append("INSERT INTO ").append(config.tableName).append(" VALUES (?");
+            for (int i = 1; i < config.numPK + config.numCols; ++i) {
+                sb.append(",?");
+            }
+            sb.append(")");
+            PreparedStatement insert = conn.prepareStatement(sb.toString());
+            Random rnd = ThreadLocalRandom.current();
+            int[] PK = new int[config.numPK];
+
+            for (;;) {
+                int idx = taskId.getAndIncrement();
+                if (idx >= batchId.length) break;
+                long mini = batchId[idx] * batchSize;
+                long maxi = Math.min(mini + batchSize, config.numRows);
+                for (long i = mini; i < maxi; ++i) {
+                    makePK(i, PK);
+                    for (int pk = 0; pk < config.numPK; ++pk) {
+                        insert.setInt(1 + pk, PK[pk]);
+                    }
+
+                    for (int col = 1; col <= config.numCols; ++col) {
+                        insert.setDouble(config.numPK + col, rnd.nextDouble());
+                    }
+                    insert.addBatch();
+                }
+                long start = System.currentTimeMillis();
+                int[] counts = insert.executeBatch();
+                insert.clearBatch();
+                long end = System.currentTimeMillis();
+                int count = 0;
+                for (int c : counts) count += c;
+                if (count != maxi - mini) {
+                    updateStats(STAT_ERROR);
+                }
+                if (count > 0) {
+                    updateStats(STAT_CREATE, count, end - start);
+                }
+            }
+
+            insert.close();
+        }
+        catch (Throwable t) {
+            LOG.error("Connection broken", t);
+            abort = true;
+        }
+    }
+
+    /*
+     *    Benchmarks create a set of prebuilt queries that select from 1 to 9 of the feature set tables selecting 5 columns from each table.
+     *    In a table with 10 columns, select columns c2, c4, c6, c8, c10, with 100 columns: c20, c40, c60, c80, c100 and with 1000: c200, c400, c600, c800, c1000.
+     *
+     *    SELECT <tableName>.<columnName>, ...
+     *    FROM <tableName>, ...
+     *    WHERE (<tableName>.<primaryKey>, ...) = (<random value>, ...)
+     *
+     *    Prepared queries are run from multiple concurrent connections for the specified number of operations per connection.
+     *    Query results are read and hashed for possible validation (not implemented).
+     *    Query execution times are collected to compute statistics provided by the general framework.
+     */
```

Comment:
good information. we might want to have this at the top of the file?

Reply:
Although it's part of the benchmark spec, provided in a separate document from which this description was borrowed, this comment really explains what _runQueries()_ does. It's similar in that regard to the _createTables()_ comment above, which explains how tables are created. I thought separating comments from what they explain would decrease clarity.